adapters/httpapi_compact.c

```diff
@@ -17,6 +17,7 @@
 #include "azure_c_shared_utility/tlsio.h"
 #include "azure_c_shared_utility/threadapi.h"
 #include "azure_c_shared_utility/shared_util_options.h"
+#include "azure_c_shared_utility/http_proxy_io.h"
```

Comment:
The unit tests for these changes are missing, but no worries, we will implement them as soon as the changes below are done. Please address the other comments and we will add unit tests on top of your changes.